tests/test_decision_service.py

```diff
@@ -1421,3 +1390,238 @@ def test_get_variation_for_feature__returns_none_for_user_in_group_experiment_no
         mock_decision.assert_called_once_with(
             self.project_config, self.project_config.get_experiment_from_id("32222"), "test_user", None, False
         )
+
+    def test_get_variation_for_feature__returns_variation_for_feature_in_mutex_group_bucket_less_than_2500(
+        self,
+    ):
+        """ Test that if a user is in the mutex group and the user bucket value should be less than 2500."""
+
+        feature = self.project_config.get_feature_from_key("test_feature_in_exclusion_group")
+        expected_experiment = self.project_config.get_experiment_from_key("group_2_exp_1")
+        expected_variation = self.project_config.get_variation_from_id(
+            "group_2_exp_1", "38901"
+        )
+        user_attr = {"experiment_attr": "group_experiment"}
+        with mock.patch(
+            'optimizely.bucketer.Bucketer._generate_bucket_value', return_value=2400) as mock_generate_bucket_value,\
+                mock.patch.object(self.project_config, 'logger') as mock_config_logging:
+
+            variation_received, _ = self.decision_service.get_variation_for_feature(
+                self.project_config, feature, "test_user", user_attr
+            )
+
+            self.assertEqual(
+                decision_service.Decision(
+                    expected_experiment,
+                    expected_variation,
+                    enums.DecisionSources.FEATURE_TEST,
+                ),
+                variation_received,
+            )
+
+        mock_config_logging.debug.assert_called_with('Assigned bucket 2400 to user with bucketing ID "test_user".')
+        mock_generate_bucket_value.assert_called_with('test_user42222')
```

Comment:
```suggestion
        mock_generate_bucket_value.assert_called_with('test_user19229')
```
group bucketing is more interesting here